### PR TITLE
Add ?version=x.x.x parameter in URL to load the version asked directly

### PR DESCRIPTION
**Pull request by tibeoh**
It's very useful to have a specific link for each version of the documentation.

Use case: I have to send links targeted the documentation website for my clients. But sometimes, they are using an older version of my API, so I have to specify in my email _"click on the top right button and select version X.X.X"_. It's not practical...

My solution:
The feature is very simple. I check if an URL param `version` exists and if the version number is known. In this case, I load the good version directly (and not the last one).

Of course, if URL param `version` does not exist, no change and last version is displayed.

Other change: If the user select an other version in the version selector (on the top right), the `version` URL param is added or updated.

It has been tested with and without anchor in URL.

Sorry for the Prettier parsing, but it was unreadable. If you want to check my changes, please check [this commit d149203](https://github.com/apidoc/apidoc/commit/d149203d06c351d4c79345647ce36234a84ea774).

I'm available if you have any question.

### CODE REVIEW

**Comment by NicolasCARPi** [comment]
Hello,

Thank you for your contribution. This project uses a 4 spaces indent for javascript files: [see .editorconfig](https://github.com/apidoc/apidoc/blob/master/.editorconfig#L13).

Would it be possible to commit only the changes without the reformating? This would also help with the diff.

**Comment by tibeoh** [comment]
Hello @NicolasCARPi 

> Would it be possible to commit only the changes without the reformating? This would also help with the diff.

It's updated!

File: template/main.js [hunk 1]
```diff
@@ -285,28 +292,30 @@ require(['jquery', 'lodash', 'locales', 'handlebarsExtended', 'apiProject', 'api
 
     // render sidenav
     var fields = {
-        nav: nav,
+        nav: nav
     };
-    $('#sidenav').append(templateSidenav(fields));
+    $('#sidenav').append( templateSidenav(fields) );
```
Comment (NicolasCARPi) [inline template/main.js]: why are spaces added here?

File: template/main.js [hunk 2]
```diff
@@ -315,27 +324,27 @@ require(['jquery', 'lodash', 'locales', 'handlebarsExtended', 'apiProject', 'api
         articleVersions[groupEntry] = {};
 
         // render all articles of a group
-        api.forEach(function (entry) {
-            if (groupEntry === entry.group) {
+        api.forEach(function(entry) {
+            if(groupEntry === entry.group) {
```
Comment (NicolasCARPi) [inline template/main.js]: there should be a space after an if

**Comment by NicolasCARPi** [comment]
While I have nothing against adding trailing commas and fixing spacing, the diff is still huge and I have no idea what you really changed.

Please make it so the diff only shows the minimum changeset pertaining to the new feature. And then you can make another PR with the prettyfied code.

**Comment by tibeoh** [comment]
You checked an outdated file. My last commit fix all the difference and respect your conventions.
https://github.com/apidoc/apidoc/pull/864/files

**Comment by NicolasCARPi** [comment]
I see it now, cheers :)

File: template/main.js [hunk 3]
```diff
@@ -14,7 +14,7 @@ require.config({
         webfontloader: './vendor/webfontloader',
         list: './vendor/list.min',
         apiData: './api_data',
-        apiProject: './api_project',
+        apiProject: './api_project'
```
Comment (NicolasCARPi) [inline template/main.js:17]: please keep trailing commas

File: template/main.js [hunk 4]
```diff
@@ -533,16 +533,53 @@ require([
...
-    // HTML-Template specific jQuery-Functions
-    //
-    // Change Main Version
-    $('#versions li.version a').on('click', function(e) {
-        e.preventDefault();
+    /*
+      Return base URL with ?param=paramVal added or updated
+    */
+    function updateURLParameter(url, param, paramVal) {
+        var TheAnchor = null;
```
Comment (NicolasCARPi) [inline template/main.js:540]: use camelCase for variables names

File: template/main.js [hunk 5]
```diff
@@ -533,16 +533,53 @@ require([
...
+                }
+            }
+        } else {
+            var tmpAnchor = baseURL.split('#');
+            var TheParams = tmpAnchor[0];
+            TheAnchor = tmpAnchor[1];
 
-        var selectedVersion = $(this).html();
-        $('#version strong').html(selectedVersion);
+            if (TheParams) baseURL = TheParams;
```
Comment (NicolasCARPi) [inline template/main.js:566]: don't use short form of if

**Comment by tibeoh** [comment]
Hi @NicolasCARPi do you need something else to merge the PR?
Do not hesitate 😉

**Comment by NicolasCARPi** [comment]
@tibeoh Yes I need to test your changes. Don't hesitate to ping me again if it's taking too long!

**Comment by NicolasCARPi** [comment]
I tried your code. If I change the version in the top right dropdown, it gets updated in the url, ok.

But if I input directly the version in the url and hit that, it resets to 0.3.0 (having 0.2.0 inputted).

**Comment by NicolasCARPi** [comment]
Closing as requested changes were not addressed.

**Pull request closed without merging**